src/hooks/useSelector.js

```diff
@@ -107,8 +107,18 @@ export function createSelectorHook(context = ReactReduxContext) {
       ? useDefaultReduxContext
       : () => useContext(context)
   return function useSelector(selector, equalityFn = refEquality) {
-    if (process.env.NODE_ENV !== 'production' && !selector) {
-      throw new Error(`You must pass a selector to useSelector`)
+    if (process.env.NODE_ENV !== 'production') {
+      if (!selector) {
+        throw new Error(`You must pass a selector to useSelector`)
+      }
+      if (typeof selector !== 'function') {
+        throw new Error(`You must pass a function as a selector to useSelector`)
+      }
+      if (typeof equalityFn !== 'function') {
+        throw new Error(
+          `You must pass a function as an equality function to useSelector`
```

Comment:
Not going to lie, I would love to add a little sass on this one:
```suggestion
          `You must pass a _function_ as an equality _function_ to useSelector...`
```
(Note, please don't merge this :P)